go/storage/fuzz/fuzz.go

```diff
@@ -0,0 +1,53 @@
+// +build gofuzz
+
+package fuzz
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature/signers/file"
+	commonFuzz "github.com/oasislabs/oasis-core/go/common/fuzz"
+	"github.com/oasislabs/oasis-core/go/common/identity"
+	"github.com/oasislabs/oasis-core/go/storage"
+)
+
+const (
+	dataDir string = "/tmp/oasis-node-fuzz-storage"
+	identityDir string = dataDir + "/identity"
+)
+
+func Fuzz(data []byte) int {
+	signerFactory := file.NewFactory(identityDir, signature.SignerNode, signature.SignerP2P, signature.SignerConsensus)
+	identity, err := identity.Load(identityDir, signerFactory)
+	if err != nil {
+		panic(err)
+	}
+
+	// Every fuzz worker should get its own database, otherwise we can't work.
+	localDB, err := ioutil.TempDir(dataDir, "worker")
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(localDB)
+
+	// Create the storage backend service.
+	storage, err := storage.New(context.Background(), localDB, identity, nil, nil)
```

Comment:
Would it be possible to only create the database once and have it shared between invocations? Or am I mistaken in how `Fuzz` is used?

Reply:
The fuzzer doesn't really provide any per-worker startup/shutdown hooks. I could use `init` for startup, but I can't see any solution for cleanup, which we'd need to properly close the database. When I tried running with just one database (and a single worker process), the library decided to stop working with a `resource temporarily unavailable`.